jib-cli/src/main/java/com/google/cloud/tools/jib/cli/cli2/Jar.java

```diff
@@ -84,7 +84,9 @@ public Integer call() {
         return 1;
       }
 
-      Containerizer containerizer = Containerizers.from(commonCliOptions, logger);
+      CacheDirectories cacheDirectories =
+          CacheDirectories.from(commonCliOptions, jarFile.getParent());
```

Comment:
Technically I think the `jarFile` path (not the parent) can still work. The difference is whether JARs in the same directory will use the same hash directory and share the app cache. Is there a specific reason to use a parent directory?

One thing matters is when that `jarFile` doesn't have a parent directory, which will be a common case, e.g., when doing `jib jar app.jar`. We learned that `jarFile.getParent() == null` in this case. So I think the argument check in `from()` will fail. 

And I just got this question if we can't just use a fixed app cache directory (without hashed sub-directories) shared by every Jib CLI build. But maybe it's good to have a hashed sub-directory per app to avoid generating too many directories in the cache root?

Reply:
> We learned that jarFile.getParent() == null in this case. 

Interesting, this is probably just because the parent path element doesn't exist, but shouldn't preclude us from getting a full path and then determining the parent?

Reply:
If the original path value (e.g., `app.jar`) can be resolved with `.getCanonicalPath()` (e.g., the file exists in the current working directory), then yeah, I think `.getParent()` on the resolved path should work. But calling `.getCanonicalPath()` after `.getParent()` will probably fail. In any case, this code will fail as it will not pass the `Preconditions.checkArgument()`.

Reply:
Right, perhaps the jar is fine then. I was initially worried that the jar would be different each time (like includes version number or something), and then we'll fail to reuse the cache.

The other option is to use a common application cache and another separate directory for "other" project files?

Reply:
> the jar would be different each time (like includes version number or something), and then we'll fail to reuse the cache.

Oh, that's a very good point.